[RGen] Bump the version of Roslyn and other nugets.

mandel-macaque opened the pull request:
Bump the version of the Roslyn nugets which allows use to check if there are any compilation warnings/errors in the code used for the tests. This simplifies the debug process when we have a mistake in the test code.

Some warnings that are incorrect from the roslyn analyzers had to be ignored, tests show that the warning is a false positive.

mandel-macaque committed:
[RGen] Fix support for WeakDelegates.

WeakDelegate properties are marched as a ArgumentSemantic.Weak which in
a normal case would mean that we do not mark them as dirty, but that is
not the case for eak delegates since we do know we provide a strong
delegate in the class. For example bgen will generate the following for
a weak delegate:

```csharp
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
object? __mt_WeakDataSource_var;
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public virtual NSObject? WeakDataSource {
	[Export ("dataSource", ArgumentSemantic.Weak)]
	get {
		global::UIKit.UIApplication.EnsureUIThread ();
		NSObject? ret;
		if (IsDirectBinding) {
			ret = Runtime.GetNSObject (global::ObjCRuntime.Messaging.NativeHandle_objc_msgSend (this.Handle, Selector.GetHandle ("dataSource")), false)!;
		} else {
			ret = Runtime.GetNSObject (global::ObjCRuntime.Messaging.NativeHandle_objc_msgSendSuper (this.SuperHandle, Selector.GetHandle ("dataSource")), false)!;
		}
		MarkDirty ();
		__mt_WeakDataSource_var = ret;
		return ret!;
	}
	[Export ("setDataSource:", ArgumentSemantic.Weak)]
	set {
		global::UIKit.UIApplication.EnsureUIThread ();
		var value__handle__ = value.GetHandle ();
		if (IsDirectBinding) {
			global::ObjCRuntime.Messaging.void_objc_msgSend_NativeHandle (this.Handle, Selector.GetHandle ("setDataSource:"), value__handle__);
		} else {
			global::ObjCRuntime.Messaging.void_objc_msgSendSuper_NativeHandle (this.SuperHandle, Selector.GetHandle ("setDataSource:"), value__handle__);
		}
		GC.KeepAlive (value);
		MarkDirty ();
		__mt_WeakDataSource_var = value;
	}
}
```
And the followign for the strong delegate:
```csharp
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public IQLPreviewControllerDataSource DataSource {
	get {
		return (WeakDataSource as IQLPreviewControllerDataSource)!;
	}
	set {
		var rvalue = value as NSObject;
		if (!(value is null) && rvalue is null)
			throw new ArgumentException ("The object passed of type " + value.GetType () + " does not derive from NSObject");
		WeakDataSource = rvalue;
	}
}
```
With this commit the same happens for rgen. This is not the complete fix
because we will need to still generate the strong delegate in rgen, that
will happen in follow up change (needs a new flag/api to mark the
property).

mandel-macaque committed:
[RGen] Generate the StrongDelegates property for a WeakDelegate.

This commit allows to generate the strong delegate for a weak delegate
in a bindings. The API provides two ways to configure the strong
property:

1. StrongDelegateType: provide a Type for the new property.
2. StrongDelegateName: Allow to choose the generated property name. If
   empty the name will be the name of the WeakDelefate with the Weak
   prefix removed.

We have some tests updates:

1. Remove the warnings from the tests. Some are due to the API being
   experimenta, other due to a rename. We will remove all errors in a
   comming PR.
2. Remove the strong delegate property from the test source since it is
   now generated.

This completes the work for support properties (so far).

Commit:
Auto-format source code

mandel-macaque committed:
Merge branch 'main' into dev/mandel/generate-strong-delegate

mandel-macaque committed:
[RGen] Bump the version of Roslyn and other nugets.

Bump the version of the Roslyn nugets which allows use to check if there
are any compilation warnings/errors in the code used for the tests. This
simplifies the debug process when we have a mistake in the test code.

Some warnings that are incorrect from the roslyn analyzers had to be
ignored, tests show that the warning is a false positive.

Commit:
Auto-format source code

vs-mobiletools-engineering-service2 commented:
# :white_check_mark: [CI Build #4a5c34d] Build passed (Build packages) :white_check_mark:




[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855299&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: build Build packages)

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build #64084dc] Build failed (Detect API changes) :fire:

Build failed for the job 'Detect API changes' (with job status 'Failed')


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855357&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [64084dc245740fbe33d8f97a32460e2ac1780bf1](https://github.com/dotnet/macios/pull/23234/commits/64084dc245740fbe33d8f97a32460e2ac1780bf1)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: build Detect API changes)

vs-mobiletools-engineering-service2 commented:
# :fire: [CI Build #64084dc] Build failed (Build packages) :fire:

Build failed for the job 'Build packages' (with job status 'Failed')


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855320&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [64084dc245740fbe33d8f97a32460e2ac1780bf1](https://github.com/dotnet/macios/pull/23234/commits/64084dc245740fbe33d8f97a32460e2ac1780bf1)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: build Build packages)

vs-mobiletools-engineering-service2 commented:
## :heavy_exclamation_mark: API diff for current PR / commit (Breaking changes)

<details><summary>.NET ( :heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark: )</summary>

* iOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/previous-api-comparison/diff/Microsoft.iOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/6e8494f0cca1f2321ec9c0b4e59978ee) ( :heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark: )
* tvOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/previous-api-comparison/diff/Microsoft.tvOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/52ade3cc6a6e944298606b768804ee25) ( :heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark: )
* MacCatalyst: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/previous-api-comparison/diff/Microsoft.MacCatalyst.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/3228361e53c7fe3a2aa2e9eb644354b5) ( :heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark: )
* macOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/previous-api-comparison/diff/Microsoft.macOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/57731c7fcc7417f265888c3df453b2b7) ( :heavy_exclamation_mark: Breaking changes :heavy_exclamation_mark: )

</details>


<!-- BreakingChangesDetected -->


## :white_check_mark: API diff vs stable

<details><summary>.NET ( No breaking changes )</summary>

* iOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/stable-api-comparison/diff/Microsoft.iOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/932372d806e16222d31b82393fbd6bff) ( No breaking changes )
* tvOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/stable-api-comparison/diff/Microsoft.tvOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/1ae88db17986499b17d244a6d8e78f15) ( No breaking changes )
* MacCatalyst: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/stable-api-comparison/diff/Microsoft.MacCatalyst.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/4a6dc4221d8269376e66133be65fec2f) ( No breaking changes )
* macOS: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/stable-api-comparison/diff/Microsoft.macOS.html) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/938c5398a028765cff89ec116350048b) ( No breaking changes )

</details>


## :information_source: Generator diff

Generator Diff: [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/generator-diff/index.html) (html) [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/macios/detected-changes/20250628.10/11855308-1/;/generator-diff/generator.diff) (raw diff) [gist](https://gist.github.com/vs-mobiletools-engineering-service2/23af94575c2afb66f678481df9ce766d) (raw diff) - Please review changes)


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=11855308&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: api diff)

vs-mobiletools-engineering-service2 commented:
# :computer: [CI Build #4a5c34d] Tests on macOS X64 - Mac Sonoma (14) passed :computer:

:white_check_mark: **All** tests on macOS X64 - Mac Sonoma (14) passed.

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855359&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: macOS X64 - Mac Sonoma _14_)

vs-mobiletools-engineering-service2 commented:
# :computer: [CI Build #4a5c34d] Tests on macOS arm64 - Mac Sequoia (15) passed :computer:

:white_check_mark: **All** tests on macOS arm64 - Mac Sequoia (15) passed.

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855359&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: macOS arm64 - Mac Sequoia _15_)

vs-mobiletools-engineering-service2 commented:
# :computer: [CI Build #4a5c34d] Tests on macOS M1 - Mac Ventura (13) passed :computer:

:white_check_mark: **All** tests on macOS M1 - Mac Ventura (13) passed.

[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11855359&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: macOS M1 - Mac Ventura _13_)

vs-mobiletools-engineering-service2 commented:
# :fire: [CI Build #4a5c34d] Test results :fire:

# Test results
:x: Tests failed on VSTS: test results

0 tests crashed, 2 tests failed, 117 tests passed.

## Failures

### :x: dotnettests tests (Multiple platforms)

<summary>1 tests failed, 0 tests passed.</summary>
<details>

 * DotNet tests: Failed (Execution failed with exit code 1)
</details>
</details>

[Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsdotnettests_multiple-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsdotnettests_multiple-1&api-version=6.0&$format=zip)

### :x: monotouch tests (macOS)

<summary>1 tests failed, 8 tests passed.</summary>
<details>

 * monotouch-test/macOS/Debug (managed static registrar): Failed (Test run failed.
Tests run: 3172 Passed: 3059 Inconclusive: 7 Failed: 1 Ignored: 112)
</details>
</details>

[Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsmonotouch_macos-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsmonotouch_macos-1&api-version=6.0&$format=zip)

## Successes

:white_check_mark: cecil: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testscecil-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testscecil-1&api-version=6.0&$format=zip)
:white_check_mark: dotnettests (iOS): All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsdotnettests_ios-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsdotnettests_ios-1&api-version=6.0&$format=zip)
:white_check_mark: dotnettests (MacCatalyst): All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsdotnettests_maccatalyst-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsdotnettests_maccatalyst-1&api-version=6.0&$format=zip)
:white_check_mark: dotnettests (macOS): All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsdotnettests_macos-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsdotnettests_macos-1&api-version=6.0&$format=zip)
:white_check_mark: dotnettests (tvOS): All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsdotnettests_tvos-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsdotnettests_tvos-1&api-version=6.0&$format=zip)
:white_check_mark: framework: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsframework-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsframework-1&api-version=6.0&$format=zip)
:white_check_mark: fsharp: All 4 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsfsharp-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsfsharp-1&api-version=6.0&$format=zip)
:white_check_mark: generator: All 5 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsgenerator-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsgenerator-1&api-version=6.0&$format=zip)
:white_check_mark: interdependent-binding-projects: All 4 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsinterdependent-binding-projects-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsinterdependent-binding-projects-1&api-version=6.0&$format=zip)
:white_check_mark: introspection: All 4 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsintrospection-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsintrospection-1&api-version=6.0&$format=zip)
:white_check_mark: linker: All 44 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testslinker-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testslinker-1&api-version=6.0&$format=zip)
:white_check_mark: monotouch (iOS): All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsmonotouch_ios-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsmonotouch_ios-1&api-version=6.0&$format=zip)
:white_check_mark: monotouch (MacCatalyst): All 15 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsmonotouch_maccatalyst-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsmonotouch_maccatalyst-1&api-version=6.0&$format=zip)
:white_check_mark: monotouch (tvOS): All 8 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsmonotouch_tvos-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsmonotouch_tvos-1&api-version=6.0&$format=zip)
:white_check_mark: msbuild: All 2 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsmsbuild-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsmsbuild-1&api-version=6.0&$format=zip)
:white_check_mark: windows: All 3 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//windows_integrationwindows-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-windows_integrationwindows-1&api-version=6.0&$format=zip)
:white_check_mark: xcframework: All 4 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsxcframework-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsxcframework-1&api-version=6.0&$format=zip)
:white_check_mark: xtro: All 1 tests passed. [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250628.10/11855359//simulator_testsxtro-1/;/tests/vsdrops_index.html) [Download](https://devdiv.visualstudio.com/DevDiv/_apis/build/builds/11855359/artifacts?artifactName=HtmlReport-simulator_testsxtro-1&api-version=6.0&$format=zip)

[comment]: <> (This is a test result report added by Azure DevOps)

[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=11855359&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [4a5c34dbcf335bfafa60511393a060d4da353610](https://github.com/dotnet/macios/pull/23234/commits/4a5c34dbcf335bfafa60511393a060d4da353610)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: test results)

mandel-macaque committed:
[RGen] Add helper methods for the TaskCompletionSource invocations. (#23235)

Create two helper methods that will allow to interact with a tcs which
is used in Async methods.

---------

Co-authored-by: GitHub Actions Autoformatter <github-actions-autoformatter@xamarin.com>

vs-mobiletools-engineering-service2 commented:
# :fire: [CI Build #de62f1b] Build failed (Build packages) :fire:

Build failed for the job 'Build packages' (with job status 'Failed')


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11864139&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [de62f1bc448f0e887e6871f9a9a6180098f5bbc6](https://github.com/dotnet/macios/pull/23234/commits/de62f1bc448f0e887e6871f9a9a6180098f5bbc6)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: build Build packages)

vs-mobiletools-engineering-service2 commented:
# :fire: [PR Build #de62f1b] Build failed (Detect API changes) :fire:

Build failed for the job 'Detect API changes' (with job status 'Failed')


[Pipeline](https://dev.azure.com/devdiv/DevDiv/_build/index?buildId=11864187&view=ms.vss-test-web.test-result-details) on Agent 
Hash: [de62f1bc448f0e887e6871f9a9a6180098f5bbc6](https://github.com/dotnet/macios/pull/23234/commits/de62f1bc448f0e887e6871f9a9a6180098f5bbc6)  [PR build]

[comment]: <> (This is a comment added by Azure DevOps, id: build Detect API changes)
